src/Framework/Analyzer/test/MinimalActions/DetectMismatchedParameterOptionalityTest.cs

```diff
@@ -0,0 +1,407 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Microsoft.AspNetCore.Analyzers.Testing.Utilities.CSharpCodeFixVerifier<
+    Microsoft.AspNetCore.Analyzers.DelegateEndpoints.DelegateEndpointAnalyzer,
+    Microsoft.AspNetCore.Analyzers.DelegateEndpoints.DelegateEndpointFixer>;
+
+namespace Microsoft.AspNetCore.Analyzers.DelegateEndpoints;
+
+public partial class DetectMismatchedParameterOptionalityTest
```

Comment:
Could you add unit tests for the tokenizer e.g. : https://github.com/dotnet/aspnetcore/pull/35732/files#diff-742e27fb1eae81bcb0bfd8113ccc306c0add53e7ee7c13b85d9c1917c974b9b5R14-R34?

Reply:
Ah forgot we talked about this! I coped over the test cases here and included a few interesting ones from https://github.com/dotnet/aspnetcore/blob/b46c5d5e15bdc6e4babd5194e0a1723d3e9cfc46/src/Http/Routing/test/UnitTests/Template/TemplateParserTests.cs